src/foundation.cs

```diff
@@ -3153,7 +3153,7 @@ public interface NSException : NSCoding, NSCopying {
 	}
 
 	public delegate void NSExpressionHandler (NSObject evaluatedObject, NSExpression [] expressions, NSMutableDictionary context);
```

Comment:
since it's unused by Xamarin.*.dll, except for obsolete API, it should be:

1. decorated with it's own `[Obsolete]` attribute, so external usage of the delegate trigger warnings (and code reviews);

2. under a `#if !XAMCORE_4_0` block so our next, _if ever_, breaking changes release can drop it;